src/chapters/methods/exercises.rst

```diff
@@ -0,0 +1,188 @@
+   How many things are being repeated?
+
+There is no starter code for these exercises, so create a new C# project
+at `repl.it <https://repl.it/login>`__ to get started.
+
+.. admonition:: Tip
+
+   As you create these methods, keep the shape of your output in mind.  
+   Remember the differences between ``Console.WriteLine`` and ``Console.Write`` .  
+
+
+Rectangles
+----------
+
+#. Write a method ``MakeLine(size)`` that returns a line with exactly ``size``
+   hashes.
+
+   .. sourcecode:: csharp
+
+      MakeLine(5);
+
+   **Console Output**
+
+   ::
+
+      #####
+
+#. Write a method called ``MakeSquare(size)`` that returns a ``size`` by
+   ``size`` iteration of hashes. 
+
+   .. sourcecode:: csharp
+
+      MakeSquare(5);
+
+   **Console Output**
+
+   ::
+
+      #####
+      #####
+      #####
+      #####
+      #####
+
+Asthetically it's not a "perfect" square, but a single argument led to its creation.
+
+#. Write a method ``MakeRectangle(width, height)`` that returns a
+   rectangle with the given width and height. 
+
+   .. sourcecode:: csharp
+
+      MakeRectangle(5, 3);
+
+   **Console Output**
+
+   ::
+
+      #####
+      #####
+      #####
+
+
+Triangles
+----------
+
+#. Write a method ``MakeDownwardStairs(height)`` that prints the staircase
+   pattern shown below, with the given height. 
+
+   .. sourcecode:: csharp
+
+      MakeDownwardStairs(5);
+
+   **Console Output**
+
+   ::
+
+      #
+      ##
+      ###
+      ####
+      #####
+
+2. Write a method ``MakeSpaceLine(numSpaces, numChars)`` that returns a line
+   with exactly the specified number of spaces, followed by the
+   specified number of hashes, followed again by ``numSpaces`` more spaces.
+
+   .. sourcecode:: csharp
+
+      MakeSpaceLine(3, 5);
+
+   **Console Output**
+
+   ::
+
+      ___#####___
+
+   .. note:: We have inserted underscores to represent spaces, so they are visible in the output. Don't do this in your code.
+
+#. Write a method ``MakeIsoscelesTriangle(height)`` that returns a triangle
+   of the given height.
+
+   .. sourcecode:: csharp
+
+      MakeIsoscelesTriangle(5);
+
+   **Console Output**
+
+   ::
+
+          #
+         ###
+        #####
+       #######
+      #########
+
+
+.. admonition:: Tip 
+
+   Consider the top line of the triangle to be level 0, the next to be line 1, and so on. 
+   Then line ``i`` is a space-line with ``height - i - 1`` spaces and ``2 * i + 1`` hashes.
+
+Diamonds
+---------
+
+#. Write a method ``MakeDiamond(height)`` that returns a diamond where the
+   triangle formed by the *top* portion has the given height.
+
+   .. sourcecode:: csharp
+
+      MakeDiamond(5);
+
+   **Console Output**
+
+   ::
+
+          #
+         ###
+        #####
+       #######
+      #########
+      #########
+       #######
+        #####
+         ###
+          #
+
+   .. tip::
+
+      Consider what happens if you create a triangle and reverse it within the same method?
```

Comment:
im not getting the idea behind this tip. can you elaborate? or maybe just get rid of it?

Reply:
This was from the JS book.  I didn't really use it when working on solutions.  I left it in case someone found it helpful.  I;m not very attached to it, so I have no issue pulling it.